src/app.js

```diff
@@ -38,3 +38,15 @@ fetch(DATA_URL)
   .catch((err) => {
     console.error(err);
   });
+
+// we want to apply some special CSS only for users who are
+// using the keyboard to navigate (e.g. focus state for custom check boxes)
+// so we add a class to the body
+function handleFirstTab(e) {
+  if (e.keyCode === 9) { // the tab key
```

Comment:
This is cool. But there's no facility to remove this state. After you tab once, you get the highlights etc. even when you use the mouse. Not a big deal but maybe mouse click could remove class & restore listener?

Reply:
This is by design. Focus rings are harmless, so the extra complexity to remove the focus ring on mouse move, and then remove the mouse listener and reattach the keydown listener, then remove it again on tab and add the mouse listener again and so on a so forth is, I think, too much complexity for too little benefit.

I wrote a [whole blog post about it](https://hackernoon.com/removing-that-ugly-focus-ring-and-keeping-it-too-6c8727fefcd2) :)